src/web/collectors/CollectorsList.jsx

```diff
@@ -145,54 +145,56 @@ const CollectorList = React.createClass({
+    const showInactiveHint = (this.state.showInactive ? null : ' and/or click on "Include inactive collectors"');
     return <Alert>There are no collectors to show. Try adjusting your search filter{showInactiveHint}.</Alert>;
   },
-  _onSubmit(event) {
-    event.preventDefault();
+  _onFilterChange(filteredRows) {
+    this.setState({ filteredRows });
   },
-  _onFilterChange(event) {
-    this.setState({ filter: event.target.value });
+  _isLoading() {
+    return !this.state.collectors;
   },
-  render() {
-    if (this.state.collectors) {
-      const collectors = this._getFilteredCollectors()
-        .filter((collector) => {
-          return (this.state.showInactive || collector.active);
-        })
-        .sort(this._bySortField)
-        .map((collector) => {
-          return <CollectorRow key={collector.id} collector={collector} />;
-        });
-
-      const showOrHideInactive = (this.state.showInactive ? 'Hide' : 'Include');
 
-      const collectorList = (collectors.length > 0 ? this._formatCollectorList(collectors) : this._formatEmptyListAlert());
+  render() {
+    if (this._isLoading()) {
+      return <Spinner />;
+    }
 
-      return (
-        <Row>
-          <Col md={12}>
-            <div className="pull-right">
-              <Button bsStyle="primary" bsSize="small" onClick={this.toggleShowInactive}>
-                {showOrHideInactive} inactive collectors
-              </Button>
-            </div>
+    const collectors = (this.state.filteredRows || this.state.collectors)
+      .filter((collector) => {
+        return (this.state.showInactive || collector.active);
+      })
+      .sort(this._bySortField)
+      .map((collector) => {
+        return <CollectorRow key={collector.id} collector={collector} />;
+      });
 
-            <form className="form-inline collectors-filter-form" onSubmit={this._onSubmit}>
-              <div className="form-group form-group-sm">
-                <label htmlFor="collectorsfilter" className="control-label">Filter collectors:</label>
-                <input type="text" name="filter" id="collectorsfilter" className="form-control"
-                       value={this.state.filter} onChange={this._onFilterChange} />
-              </div>
-            </form>
+    const showOrHideInactive = (this.state.showInactive ? 'Hide' : 'Include');
 
-            {collectorList}
-          </Col>
-        </Row>
-      );
-    }
+    const collectorList = (collectors.length > 0 ? this._formatCollectorList(collectors) : this._formatEmptyListAlert());
 
-    return <Spinner />;
+    return (
+      <Row>
+        <Col md={12}>
+          <div className="pull-right">
+            <Button bsStyle="primary" bsSize="small" onClick={this.toggleShowInactive}>
+              {showOrHideInactive} inactive collectors
+            </Button>
+          </div>
+          <div className="form-inline collectors-filter-form">
+            <CollectorFilter ref="collectorsFilter"
```

Comment:
We didn't know about this when creating the PR, but we should avoid string refs:
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md

The easiest way to fix it is to replace the `ref` with something like: `ref={(c) => { this._collectorsFilter = c; }}`, and then use `this._collectorsFilter` when accessing the component.